core/src/main/java/org/apache/iceberg/rest/UncheckedRestException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import com.google.errorprone.annotations.FormatMethod;
+import java.util.Objects;
+
+/**
+ * Exception thrown by the REST client and catalog, to wrap a checked exception
+ * in an unchecked exception.
+ */
+public class UncheckedRestException extends RuntimeException {
```

Comment:
Can't the other rest exception handle checked exceptions?

Reply:
Absolutely. Good catch. I had a few different branches where I was working out some possible ideas, and somehow I managed to leave both of those in.

I'll remove this one. 👍